driver/CMakeLists.txt

```diff
@@ -26,6 +27,7 @@ else()
 	set(kmod_min_kver_map_aarch64 3.16)
 	set(kmod_min_kver_map_s390x 2.6)
 	set(kmod_min_kver_map_riscv64 5.0)
+	set(kmod_min_kver_map_ppc64le 2.6)
```

Comment:
Like for bpf, kmod too needs some small work:
* https://github.com/falcosecurity/libs/blob/master/driver/main.c#L2144
* https://github.com/falcosecurity/libs/blob/master/driver/ppm.h#L81

Reply:
@FedeDP I was going through this https://github.com/falcosecurity/libs/blob/c2fd3086ff1bd0fce20a24a6da9f03c47332d68a/driver/feature_gates.h#L64
We need this patch for ppc64 too?

Reply:
I am not sure! That depends if `ppc64le` is hit by the kernel behavior of not sending the clone exit event for the child; at the moment, all supported architectures but x86_64 are hit.
We have drivers tests for it though, eg: https://github.com/falcosecurity/libs/blob/master/test/drivers/test_suites/syscall_exit_suite/clone3_x.cpp#L191
So, if this test is passing fine running on ppc64le, we should not need the CAPTURE_SCHED_PROC_FORK enabled.
The same goes for CAPTURE_SCHED_PROC_EXEC.

cc @Andreagit97

Reply:
uhm it seems like `ppc64le` behaves like `x86` since looking at failed tests here https://github.com/falcosecurity/libs/pull/1475#issuecomment-1803296626 `execve` and `clone` test are passing. 

BTW always looking at the tests there is another issue:
```
[  FAILED  ] SyscallExit.execveatX_correct_exit
[  FAILED  ] SyscallExit.execveatX_execve_exit
[  FAILED  ] SyscallExit.execveatX_success_memfd
```
it seems that `ppc64le` correctly returns an `execveat` exit event when an `execveat` syscall is called (like s390x and riscv)

https://github.com/falcosecurity/libs/blob/c2fd3086ff1bd0fce20a24a6da9f03c47332d68a/test/drivers/test_suites/syscall_exit_suite/execveat_x.cpp#L214

Other architectures like x86 instead return an `execve` instead of an `execveat` exit event.

Reply:
> but should that be done with a separate PR?

We can do it in this one, since it is implementing support for kmod and old ebpf too :)

Reply:
Also please make sure to share drivers_test results on kmod,ebpf and modern ebpf once this is ready, thank you!

Reply:
Sure, will do.

Reply:
I am trying to run drivers_test for ebpf probe but I am getting this problem

```
-----------------------------------------------------
- Setup phase
-----------------------------------------------------

* Configure BPF probe tests! Probe path: /root/afsan/libs/build/driver/bpf/probe.o
* Using buffer dim: 8388608
-- BEGIN PROG LOAD LOG --
insn 6910 cannot be patched due to 16-bit range
processed 40396 insns (limit 1000000) max_states_per_insn 1 total_states 4057 peak_states 4057 mark_read 73

-- END PROG LOAD LOG --
Unable to open the engine: libscap: bpf_load_program() event=raw_tracepoint/filler/sys_procexit_e: Operation not permitted

```
Running this on a RHEL 9.2 machine.
Kernel version: `5.14.0-284.25.1.el9_2.ppc64le`

Steps I have followed:

1. `cmake -DBUILD_BPF=ON -DBPF_DEBUG=ON -DUSE_BUNDLED_DEPS=On -DENABLE_DRIVERS_TESTS=On -DBUILD_LIBSCAP_GVISOR=Off -DCREATE_TEST_TARGETS=On ..`
2. `sudo ./test/drivers/drivers_test -b`

Kmod and modern-bpf tests work fine.

Reply:
That's interesting; it is an ebpf verifier failure that we never encountered; the issue comes from https://android.googlesource.com/kernel/common/+/4e2c7b297431/kernel/bpf/verifier.c#10490 but we will need to investigate further what's happening.

Reply:
Btw i think you can open a bug report about the issue, but we can go on with this.

Reply:
Yes. I was digging around this but no luck so far. https://github.com/falcosecurity/libs/blob/041477e3acbf3ab79799bfbf28c259fde8ae7e52/driver/bpf/fillers.h#L5203
I will create a bug report for now.

Reply:
If you want to give a shot, you can try to comment the 
```
struct list_head *head = &(task->children);
	struct list_head *next_child = (struct list_head *)_READ(head->next);
	if(next_child != head)
	{
		/* We have at least one child, so we need a reaper for it */
		reaper_pid = find_new_reaper_pid(data, task);
	}
``` 
lines and check if it works!

Reply:
Tests are running now. Is `*next_child` pointing to NULL?

Reply:
Logs for all three drivers test.

[kmod_tests.log](https://github.com/falcosecurity/libs/files/13486720/kmod_tests.log)
[core_bpf_tests.log](https://github.com/falcosecurity/libs/files/13486721/core_bpf_tests.log)
[bpf_tests.log](https://github.com/falcosecurity/libs/files/13486722/bpf_tests.log)

Reply:
> Tests are running now. Is *next_child pointing to NULL?

@Andreagit97 any idea?
@mdafsanhossain can you try uncommenting that piece of code and commenting only the `reaper_pid = find_new_reaper_pid(data, task);` line too?

Reply:
Tried it. That seems to be the offending line

Reply:
Uhm it seems strange

```c
	struct list_head *head = &(task->children);
	struct list_head *next_child = (struct list_head *)_READ(head->next);
	if(next_child != head)
	{
		/* We have at least one child, so we need a reaper for it */
		reaper_pid = find_new_reaper_pid(data, task);
	}
```

if `head!= null`, `next_child` cannot be `null` because in the worst case, it would be `next_child==head`, BTW we never dereference `next_child` so it shouldn't be an issue at all :thinking:

Reply:
Yep the issue is within `find_new_reaper_pid` :/

Reply:
I opened the issue to track the legacy bpf verifier failure: https://github.com/falcosecurity/libs/issues/1521